front/lib/resources/labs_transcripts_resource.ts

```diff
@@ -0,0 +1,283 @@
+    LabsTranscriptsConfigurationModel;
+
+  constructor(blob: Attributes<LabsTranscriptsConfigurationModel>) {
+    super(LabsTranscriptsConfigurationModel, blob);
+  }
+
+  static async makeNew(
+    blob: Omit<
+      CreationAttributes<LabsTranscriptsConfigurationModel>,
+      "isActive"
+    >
+  ): Promise<LabsTranscriptsConfigurationResource> {
+    const { userId, connectionId, provider } = blob;
+    const hasExistingConfiguration =
+      await LabsTranscriptsConfigurationModel.count({
+        where: {
+          userId,
+          connectionId,
+          provider,
+        },
+      });
+    if (hasExistingConfiguration) {
+      throw new Error(
+        `A Solution configuration already exists for user ${userId} with connectionId ${connectionId} and provider ${provider}`
+      );
+    }
+    const configuration = await LabsTranscriptsConfigurationModel.create({
+      userId,
+      connectionId,
+      provider,
+      isActive: false,
+    });
+
+    return new LabsTranscriptsConfigurationResource(configuration.get());
+  }
+
+  static async findByUserIdAndProvider({
+    userId,
+    provider,
+  }: {
+    userId: ModelId;
+    provider: LabsTranscriptsProviderType;
+  }): Promise<LabsTranscriptsConfigurationResource | null> {
+    const configuration = await LabsTranscriptsConfigurationModel.findOne({
+      where: {
+        userId,
+        provider,
+      },
+    });
+
+    return configuration
+      ? new LabsTranscriptsConfigurationResource(configuration.get())
+      : null;
+  }
+
+  async setAgentConfigurationId({
+    agentConfigurationId,
+  }: {
+    agentConfigurationId: string | null;
+  }): Promise<
+    Result<
+      void,
+      | {
+          type: "not_found";
+        }
+      | Error
+    >
+  > {
+    if (this.agentConfigurationId === agentConfigurationId) {
+      return new Ok(undefined);
+    }
+
+    try {
+      await LabsTranscriptsConfigurationModel.update(
+        { agentConfigurationId },
+        {
+          where: {
+            id: this.id,
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async setEmailToNotify({
+    emailToNotify,
+  }: {
+    emailToNotify: string | null;
+  }): Promise<
+    Result<
+      void,
+      | {
+          type: "not_found";
+        }
+      | Error
+    >
+  > {
+    if (this.emailToNotify === emailToNotify) {
+      return new Ok(undefined);
+    }
+
+    try {
+      await LabsTranscriptsConfigurationModel.update(
+        { emailToNotify },
+        {
+          where: {
+            id: this.id,
+          },
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  static async getIsActive({
+    userId,
+    provider,
+  }: {
+    userId: ModelId;
+    provider: LabsTranscriptsProviderType;
+  }): Promise<boolean> {
+    const configuration = await this.findByUserIdAndProvider({
```

Comment:
All the fetch should happen outside of those functions and they should be class members.